examples/mcuboot/ota/Kconfig

```diff
@@ -0,0 +1,56 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+menuconfig EXAMPLES_MCUBOOT_OTA
```

Comment:
What's the reason for changing the application name?

Reply:
Originally those are two apps inside boot/mcuboot with two distinct names that makes sense inside that scope (MCUBOOT_UPDATE_AGENT_EXAMPLE + MCUBOOT_SLOT_CONFIRM_EXAMPLE). 

With this refactor prefix changed to `EXAMPLES_MCUBOOT_OTA` which represents examples/mcuboot/ota were OTA became the example name, which gave more sense of consistency. The suffix APP_AGENT + APP_CONFIRM are enough to represent each application inside this demo.

Do you prefer `EXAMPLES_MCUBOOT_OTA_UPDATE_AGENT` and `EXAMPLES_MCUBOOT_OTA_SLOT_CONFIRM` ?

Reply:
I don't think that OTA should be part of the name, since the update agent example does not restrict or assumes that the update will take place over a wireless network. And OTA does not imply "update".

Reply:
In fact, I'd suggest for this PR to just restrict the scope to moving the contents without performing further changes. These minor changes are difficult to track, since GitHub simply identifies the destination as a new file.

Reply:
I'll rework PR as a sequence, changes are not superfluous. The simple fact that this refactor is necessary shows that MCUboot, as is, have inconsistencies. This helps to improve into a more generic and easy to use way. Since it is required replace strings in the main NuttX repo, it is better perform once with right value, instead twice.

Maybe `EXAMPLES_MCUBOOT_UPDATE` is a more consistent prefix name and `_AGENT/_CONFIRM` better suffixes.

Reply:
>  The simple fact that this refactor is necessary 

If this is indeed necessary, the motivation for this refactor should be stated on the PR description. So far, I still consider these changes superfluous. The changes to the names and help text of Kconfig options, in my opinion, bring no improvement over the previous version.

> This helps to improve into a more generic and easy to use way. Since it is required replace strings in the main NuttX repo, it is better perform once with right value, instead twice.

How does these changes in this PR cope with it?

> Maybe EXAMPLES_MCUBOOT_UPDATE is a more consistent prefix name and _AGENT/_CONFIRM better suffixes.

The prefix is just `EXAMPLES_MCUBOOT_`. I don't understand how this notion of suffix fits here. There is no suffix in the Kconfig names.

Reply:
> If this is indeed necessary, the motivation for this refactor should be stated on the PR description. So far, I still consider these changes superfluous. The changes to the names and help text of Kconfig options, in my opinion, bring no improvement over the previous version.

_The current version of MCUboot have core and [define] example[s] together. This moves samples to apps/examples directory to clean up._ This is a clear motivation. The Kconfig name changes are necessary to keep name convention consistent once in all examples the Kconfig variables starts with EXAMPLES_.

> How does these changes in this PR cope with it?

The cleaning up require Kconfig name changes to keep consistency. This should be reflected into boards/configs/defconfig, which are related to MCUboot. README files may be affected because can describe how build/flash for users, for instance.

> The prefix is just EXAMPLES_MCUBOOT_. I don't understand how this notion of suffix fits here. There is no suffix in the Kconfig names.

The prefix is just the constant part of Kconfig symbol, which repeat in the whole Kconfig file example, nothing more. The suffix is the remaining part.

This PR moves MCUboot _samples_ to _a new application example_ inside examples/mcuboot/ota. For this initial proposal, prefix should be EXAMPLES_MCUBOOT_OTA. If this name is not OK I would like to change it. What do you suggest?

Reply:
We're not talking about the same issues here.
Regarding the code move, I am already okay with that, as I stated in my last comment. Also with the addition of `EXAMPLES_` to the Kconfig config names, which is related to the code move.
So, the previous `MCUBOOT_UPDATE_AGENT_EXAMPLE` key, once moved, should simply become `EXAMPLES_MCUBOOT_UPDATE_AGENT`. And that's it. That achieves the purpose of this PR.
That's why I said to restrict the PR to these changes only, to avoid hindering the merge.

I am not okay with everything else in this PR, because those other changes lack background.

Reply:
Ok, good. It is necessary define a name for this App. I proposed `ota`, which seems not be a good one. So, I'm asking, what in your opinion should be a good App name for this App?

Reply:
First: Why Update Agent is not a good name for the app?

Reply:
Sorry, I missed one point that you commented earlier which is important to the discussion:

> The suffix APP_AGENT + APP_CONFIRM are enough to represent each application inside this demo.

Although this demo you are proposing includes both example applications, I believe there is no need to reflect it in the folder structure or in the kconfig hierarchy. The demo is best characterized by the set of options included in the `defconfig` file.
So all the MCUboot example application should appear on the level at `examples/mcuboot`.